packages/swing-store/src/transcriptStore.js

```diff
@@ -129,13 +185,26 @@ export function makeTranscriptStore(
     ORDER BY startPos
   `);
 
+  function* dumpTranscriptItems(vatID, startPos, endPos) {
```

Comment:
Can we make it more clear this can only dump the items of a single span ?

Reply:
Yeah, that kind of bothered me too.  The issue is that if the items are uncompressed the function can dump any range of start and end positions, but it is limited to a single (entire) span if they're compressed.  So in the case of a compressed span the function name is a lie.  You _could_ write a more complicated function that would dump any range even when compressed, but there's no use case for it, the code that calls this function (which is strictly internal) wouldn't use it, it would be needlessly complicated to implement, and needlessly expensive to execute.  On the other hand, renaming it to indicate that it handles a single span only would be a lie in the case of uncompressed items.  On the other other hand, I think one of those lies is more benign than the other one -- if the name says "one span only", the caller ends up writing their own logic to dump multiple spans, which will work fine even if it introduces more calls than strictly required, but if the name implies "any position range" and then the caller uses it across span boundaries it blows up.  Plus, I suppose, the caller might not be in a position to actually know if the items are compressed or not, so they really should implement the multiple span logic anyway just as a defensive measure -- which in fact the two extant callers of this function actually do.  OK, so I've talked myself into the second lie.  I'll rename the function.  I suppose it would also make sense in that case to add an assert to ensure that the parameters actually indicate a single whole span, in which case the name wouldn't be a lie.